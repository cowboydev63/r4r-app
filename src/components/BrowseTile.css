.browse__tile {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    background: #169bd5;
    color: white;
    width: 150px;
    height: 100px;
    user-select: none;
    cursor: pointer;

    box-shadow: 0 15px 5px -12px rgba(0,0,0,0.2);
}

.browse__tile:hover {
    box-shadow: 0 18px 15px -15px rgba(0,0,0,0.2);
}

.browse__tile:active {
    box-shadow: 0 10px 5px -8px rgba(0,0,0,0.2);
}