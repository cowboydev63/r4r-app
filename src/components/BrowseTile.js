import React from 'react';
import PropTypes from 'prop-types';
import { Theme } from '../theme';
import './BrowseTile.css';

class BrowseTile extends React.PureComponent {
    static propTypes = {
        className: PropTypes.string,
        onClick: PropTypes.func.isRequired,
        onKeyPress: PropTypes.func.isRequired,
    }

    static defaultProps = {
        onClick: () => {},
        onKeyPress: () => {},
    }

    render() {
        return (
            <Theme
                type="div"
                className="browse__tile"
                role="link"
                tabIndex='0' 
                onClick={ this.props.onClick }
                onKeyPress={ this.props.onKeyPress }
            >
                <p>{ this.props.label }</p>
            </Theme>
        )
    }
}

export default BrowseTile;