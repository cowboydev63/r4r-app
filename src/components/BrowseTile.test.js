import React from 'react';
import { shallow } from 'enzyme';
import BrowseTile from './BrowseTile';

it('renders without crashing', () => {
  shallow(<BrowseTile />);
});