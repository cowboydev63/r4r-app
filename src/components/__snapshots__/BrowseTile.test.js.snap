// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseTile Component renders correctly 1`] = `
<Theme
  className="browse__tile"
  element="a"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="link"
  tabIndex="0"
>
  Test
   (4)
</Theme>
`;
